### DISCUSSION TITLE
CORS Policy Issue between Pimcore and Nuxt.js

### DISCUSSION THREAD

**Discussion by vesselin-ivanov** [pimcore/pimcore#15443] [Q&A / Support]
Dear Developers,

I hope this message finds you well. I am writing to seek assistance in resolving a CORS (Cross-Origin Resource Sharing) policy issue that I am encountering between Pimcore and Nuxt.js. I have provided details about the problem and the relevant code below.

In my Nuxt.js application, I am making a GraphQL query to a Pimcore backend to fetch some data. Here is the GraphQL query I am using:

```
const query = gql`
query {
  getNewsListing {
    edges {
      node {
        Title
      }
    }
  }
}
`

const { data } = await useAsyncQuery(query)
```

Additionally, I have configured Nuxt.js with the following settings in the nuxt.config.ts file:

```
import { defineNuxtConfig } from 'nuxt/config'

export default defineNuxtConfig({
  modules: [
    '@nuxtjs/apollo'
  ],
  apollo: {
    clients: {
      default: {
        httpEndpoint: 'http://localhost/pimcore-graphql-webservices/web',
        httpLinkOptions: {
          headers: {
            'apikey': 'fe4c45dede0bdadfbf0c242abe12a33f'
          }
        }
      }
    }
  }
})
```

Unfortunately, I am facing a CORS policy issue when making the request from Nuxt.js to Pimcore. I understand that this issue can be resolved by configuring the appropriate CORS headers on the server-side, but I don't now exactly where to add them.

I kindly request your guidance in configuring the necessary CORS headers for Pimcore. The frontend is running on http://localhost:3000, and the backend URL is http://localhost.

I would greatly appreciate any assistance or insights you can provide to help resolve this CORS policy issue. If you require any further information or code samples, please let me know, and I will be happy to provide them.

Thank you for your attention to this matter.

Best regards,
Veselin

**Comment by kingjia90** [comment]
It doesn't seem like a problem that can be solved on Pimcore side.
Please have a look to https://github.com/pimcore/pimcore/discussions/10687#discussioncomment-2416806 and eventually https://stackoverflow.com/a/10892392/2119584
